src/components/company/CompanyAddForm.tsx

```diff
@@ -0,0 +1,310 @@
+import React from 'react';
+
+import {
+    Box,
+    Text,
+    FormControl,
+    FormLabel,
+    Input,
+    Textarea,
+    Switch,
+    SimpleGrid,
+    Button,
+    GridItem
+} from '@chakra-ui/react';
+
+import { HelperText } from '@components/common';
+
+import { useValidationHelper } from 'hooks';
+import { useCompanyHelper } from './useCompanyHelper';
+
+import { ErrorMsg, RegExUtil } from 'utils';
+import type {
+    CompanyFormProps,
+    FormErrorProps,
+    ValidationMapProps
+} from 'interfaces';
+import {
+    DEFAULT_COMPANY_ADDRESS,
+    DEFAULT_COMPANY_SETTINGS,
+    DEFAULT_LMS_SETTINGS
+} from 'Constants';
+
+interface CompanyFieldProps {
+    fieldName: keyof Omit<
+        CompanyFormProps,
+        'settings' | 'governmentIdentifiers'
+    >;
+    label: string;
+    placeholder: string;
+    type: 'text' | 'textArea';
+    isRequired: boolean;
+    errorMsg: string;
+    helperText?: string;
+}
+
+const validationMap: ValidationMapProps = {
+    companyId: (companyId: string) => RegExUtil.isId(companyId),
+    name: (name: string) => RegExUtil.isName(name),
+    description: (description: string) => RegExUtil.isDescription(description),
+    rawAddress: (address: string) => RegExUtil.isDescription(address),
+    email: (email: string) => RegExUtil.isEmail(email),
+    mobileNumber: (mobile: string) => RegExUtil.isMobileNumber(mobile)
+};
+
+const requiredFields: (keyof CompanyFormProps)[] = [
+    'companyId',
+    'name',
+    'description',
+    'rawAddress',
+    'email',
+    'mobileNumber'
+];
+
+const formErrorStateInitialValues: FormErrorProps<
+    Omit<CompanyFormProps, 'settings' | 'governmentIdentifiers'>
+> = {
+    companyId: false,
+    name: false,
+    description: false,
+    rawAddress: false,
+    email: false,
+    mobileNumber: false
+};
+
+export const CompanyAddForm = () => {
+    const { hasFormFieldError, getFormErrorData } =
+        useValidationHelper(validationMap);
+    const { generateRandomGSTIN } = useCompanyHelper();
+
+    const companyFormInitialState = React.useMemo(
+        () => ({
+            companyId: '',
+            name: '',
+            description: '',
+            rawAddress: DEFAULT_COMPANY_ADDRESS,
+            email: '',
+            mobileNumber: '',
+            governmentIdentifiers: {
+                gstin: generateRandomGSTIN()
+            },
+            settings: { ...DEFAULT_COMPANY_SETTINGS }
+        }),
+        [generateRandomGSTIN]
+    );
+
+    const [form, setForm] = React.useState<CompanyFormProps>({
+        ...companyFormInitialState
+    });
+    const [formErrorState, setFormErrorState] = React.useState({
+        ...formErrorStateInitialValues
+    });
+
+    const companyFields: CompanyFieldProps[] = React.useMemo(
```

Comment:
move this into helper?

Reply:
removed this code. This was being used to create fields in the forms.